sdk-actors/src/main/java/io/dapr/actors/runtime/DaprStateAsyncProvider.java

```diff
@@ -56,7 +60,7 @@
   DaprStateAsyncProvider(DaprClient daprClient, DaprObjectSerializer stateSerializer) {
     this.daprClient = daprClient;
     this.stateSerializer = stateSerializer;
-    this.isStateSerializerDefault = stateSerializer.getClass() == DefaultObjectSerializer.class;
+    this.isStateSerializerJson = JSON_CONTENT_TYPE.equals(stateSerializer.getContentType());

```

Comment:
We can make this check even more extensible by having a deeper understanding of different media types. This change is already an important improvement.